spec/support/shared_examples.rb

```diff
@@ -1,46 +1,84 @@
 # frozen_string_literal: true
 
-# Credit: https://github.com/datariot/omniauth-paypal/blob/master/spec/support/shared_examples.rb
-# NOTE it would be useful if this lived in omniauth-oauth2 eventually
-
-shared_examples 'an oauth2 strategy' do
-  describe '#client' do
-    it 'is initialized with symbolized client_options' do
-      @options = { client_options: { 'authorize_url' => 'https://example.com' } }
+RSpec.shared_examples 'basic data retrieval tests' do
+  context 'with successful API response' do
+    before do
+      allow(access_token).to receive(:get)
+        .with('https://api.mlh.com/v4/users/me')
+        .and_return(instance_double(OAuth2::Response, parsed: { 'data' => base_user_data }))
```

Comment:
The alignment seems to be a bit different than the pattern we use, we should probably use MLH rubocop: https://github.com/MLH/MLH-Rubocop-Config

This probably should be handled in another PR though.

Reply:
Yeah, the rubocop linter config we have here passes but we may want another PR to update the whole thing to MLH's new rubocop config